### PR TITLE
[AsmPrinter] Don't check for inlineasm dialect on non-X86 platforms

### PR DESCRIPTION
**Pull request by dtellenbach**
AArch64 uses MCAsmInfo::AssemblerDialect to control the style of emitted Neon assembly. E.g. Apple platforms use AsmWriterVariantTy::Apple by default which collides with InlineAsm::AD_Intel (both value 1). Checking for inlineasm dialects on non-X86 platforms can thus lead to problems.

### CODE REVIEW

File: clang/test/CodeGen/AArch64/inline-asm-ios.c [hunk 1]
```diff
@@ -0,0 +1,23 @@
+// REQUIRES: aarch64-registered-target
```
Comment (jroelofs) [inline clang/test/CodeGen/AArch64/inline-asm-ios.c:1]: Is this a new folder? There should be a `lit.local.cfg` in there that takes care of this:

```
if "AArch64" not in config.root.targets:
    config.unsupported = True
```

If not, please add one.

Comment (dtellenbach) [reply clang/test/CodeGen/AArch64/inline-asm-ios.c]: Yep, seems like I accidentally added a new folder. Targets other than AArch64 and Arm use a subfolder but still rely on `REQUIRES`. AArch64 just dumps everything into the top-level test folder and thus have to rely on `REQUIRES`. 

I now move the test up to `test/CodeGen` and stick with `REQUIRES`. Should be reworked but not in this patch.

**Comment by dtellenbach** [comment]
Linux test failure seems unrelated (fails to build some libcxx stuff). The new test passes:

```
PASS: Clang :: CodeGen/aarch64-inlineasm-ios.c (14316 of 95454)
```

**Review by jroelofs — APPROVED** [review]

**Comment by MaskRay** [comment]
LGTM

**Pull request merged**